addons/account/wizard/account_move_send.py

```diff
@@ -431,6 +431,18 @@ def _get_mail_params(self, move):
         if self.mode == 'invoice_single':
             # We must ensure the newly created PDF are added. At this point, the PDF has been generated but not added
             # to 'mail_attachments_widget'.
+            if self.mail_attachments_widget:
+                attachments_to = []
+                for json in self.mail_attachments_widget:
+                    if isinstance(json['id'], int):
+                        attachment = self.env["ir.attachment"].search_count([('id', '=', json['id'])])
+                        if attachment:
+                            attachments_to.append(json)
+                    else:
+                        attachments_to.append(json)
+
+                self.mail_attachments_widget = attachments_to
```

Comment:
The method name is `_get_mail_params`, so it's not supposed to make changes in `self`